state/charm.go

```diff
@@ -221,6 +231,13 @@ func updateCharmOps(mb modelBackend, info CharmInfo, assert bson.D) ([]txn.Op, e
 		{"pendingupload", false},
 		{"placeholder", false},
 	}
+	if featureflag.Enabled(feature.LXDProfile) {
+		lpc, ok := info.Charm.(charm.LXDProfiler)
+		if !ok {
+			return nil, errors.New("charm doesn't have LXDCharmProfile()")
```

Comment:
I think if the `info.Charm` doesn't have a LXDProfile method, we should just ignore it and assume that it doesn't need to store the profile in the document.

Reply:
I *think* I agree with Simon here. Maybe best to accept the old interface too - there can be many Charm implementations, potentially, and they don't all have to implement LXDProfile.

Reply:
Also, do you actually want an lxd-profile field in the document when LXDProfile returns nil? (that might be best from a backward-compatibility point of view, because null is different from not-exists in mongo)